.github/workflows/unit-integration-tests.yml

```diff
@@ -201,11 +203,16 @@ jobs:
         ports:
           - 4566:4566
     steps:
+      - name: Restore repo cache
+        uses: actions/cache@0400d5f644dc74513175e3cd8d07132dd4860809 # v4.2.4
+        with:
+          path: .git
+          key: gitdb-${{ github.repository_id }}-${{ github.sha }}
       - name: Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          ref: ${{ inputs.ref || github.ref }}
-          fetch-depth: $(( ${{ github.event_name == 'pull_request' && github.event.pull_request.commits || 0 }} + 1 ))
```

Comment:
By default `fetch-depth` is `1`. This expression seems to end with `1` or `2`, which I don't think it's required.